app/Models/Transaction.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Models;
+
+use App\Model;
+use DateTime;
+use Exception;
+
+class Transaction extends Model
+{
+    protected int $id;
+    public DateTime $date;
+    public string $check;
+    public string $description;
+    public float $amount;
+
+
+    public function all(): TransactionCollection
+    {
+        $query = 'select id, date, check_number, description, amount from transactions';
+
+        $statement = $this->db->query($query);
+
+        $transactions = [];
+
+        while (($row = $statement->fetch()) !== false) {
+            $transactions[] = $this->createFromRow($row);
+        }
+
+        return new TransactionCollection($transactions);
+    }
+
+    public function save(): bool
+    {
+        $query = 'insert into transactions (date, check_number, description, amount) 
+                    values (:date, :check_number, :description, :amount)';
+
+        $statement = $this->db->prepare($query);
+        $statement->execute([
+            ':date' => $this->date->format('Y-m-d'),
+            ':check_number' => $this->check,
+            ':description' => $this->description,
+            ':amount' => $this->amount,
+        ]);
+
+        $this->id = (int)$this->db->lastInsertId();
+
+        return $this->id > 0;
+    }
+
+    public function saveAll(TransactionCollection $transactionCollection): bool
+    {
+        $query = 'insert into transactions (date, check_number, description, amount) 
+                    values (:date, :check_number, :description, :amount)';
+
+        if ($this->db->inTransaction()) {
+            throw new \PDOException('Uma transação já está em andamento');
+        }
+
+        try {
+            $this->db->beginTransaction();
+
+            $statement = $this->db->prepare($query);
+
+            /** @var Transaction $transaction */
+            foreach ($transactionCollection as $transaction) {
+                $statement->execute([
+                    ':date' => $transaction->date->format('Y-m-d'),
+                    ':check_number' => $transaction->check,
+                    ':description' => $transaction->description,
+                    ':amount' => $transaction->amount,
+                ]);
+
+                $transaction->id = (int)$this->db->lastInsertId();
+            }
+
+            return $this->db->commit();
+        } catch(\PDOException $e) {
+            $this->db->rollBack();
+            throw $e;
+        }
+    }
+
+    public static function processUploadFile(string $path): void
```

Comment:
This method probably belongs elsewhere. Maybe another class. Right now your Transaction model is responsible for too many things